cpp/log/cert.cc

```diff
@@ -966,15 +971,20 @@ util::Status Cert::IsValidNameConstrainedIntermediateCa() const {
   // If it's not a CA cert or there is no name constraint extension then we
   // don't need to apply the rules any further
   StatusOr<bool> has_ca_constraint = HasBasicConstraintCATrue();
+  StatusOr<bool> has_name_constraints = HasExtension(NID_name_constraints);
+
   if (!has_ca_constraint.ok() || !has_ca_constraint.ValueOrDie()
-      || HasExtension(NID_name_constraints) == Cert::FALSE) {
+      || !has_name_constraints.ok() || !has_name_constraints.ValueOrDie()) {
     return util::Status::OK;
   }
 
   // So there now must be a CT extension and the name constraint must not be
   // in error
-  if (HasExtension(NID_name_constraints) != Cert::TRUE
-      || HasExtension(NID_ctNameConstraintNologIntermediateCa) != Cert::TRUE) {
+  StatusOr<bool> has_ct_nolog_intermediate =
+      HasExtension(NID_ctNameConstraintNologIntermediateCa);
+
+  if (!has_name_constraints.ValueOrDie() || !has_ct_nolog_intermediate.ok() ||
```

Comment:
I think that `has_name_constraints` must be `true` here, so maybe pull this out of the `if` and `CHECK(has_name_contraints.ValueOrDie());` ?